prime-dotnet-webapi/Extensions/EnumExtensions.cs

```diff
@@ -15,5 +15,29 @@ public static class EnumExtensions
                 .OfType<TAttribute>()
                 .SingleOrDefault();
         }
+
+        // To avoid "The annotation for nullable reference types should only be used in code within a '#nullable' annotations context." warning
+#nullable enable
+        public static bool IsValid(this Enum? value)
+#nullable disable
```

Comment:
Rather than using `#pragmas` to hide warnings, it is better to find out why the warnings are happening and fix them.  In this case, its because the base class `Enum` is handled slightly differently than the enums we define. 

`Enum` is an abstract _class_:
![image](https://user-images.githubusercontent.com/39168456/151617183-ffe3acd5-b751-4df1-a3fc-d35cfc21007e.png)
which means that `Enum` can already be null. Adding the `?` annotation on a class is the nullable reference types feature that is available in newer versions of C#, hence the warning.

When we define our own enums, such as `AgreementType`, they are concrete implementations of the `Enum` abstract class. In C#, concrete implementations of the Enum class (as declared e.g `public enum AgreementType`) are _structs_ which means they are not nullable by default. So when using a concrete enum that you want nullable, that is when you have to use the `AgreementType?` syntax. 99% of the time you are using concrete enums and not manipulating the `Enum` class directly.

The fix here is to remove the question mark.

Reply:
Does this comment https://github.com/bcgov/moh-prime/pull/1963/files#r794957970 address?

Reply:
![image](https://user-images.githubusercontent.com/39168456/151638063-2bf7ef21-5bbe-4ea2-8b8a-f1fa4d9ccd16.png)
I do not see how that comment address this. The issue is that you are using ` #nullable enable` and ` #nullable disable` to suppress warnings that are informing you that you are miss-using the nullable types feature.

Reply:
Never mind, I thought the compiler was making me use a nullable type at one point